support-services/src/main/scala/com/gu/support/pricing/PriceSummaryService.scala

```diff
@@ -0,0 +1,95 @@
+package com.gu.support.pricing
+
+import com.gu.i18n.CountryGroup
+import com.gu.support.catalog._
+import com.gu.support.pricing.PriceSummaryService.getDiscountedPrice
+import com.gu.support.promotions._
+import com.gu.support.workers.BillingPeriod
+
+import scala.math.BigDecimal.RoundingMode
+
+class PriceSummaryService(promotionService: PromotionService, catalogService: CatalogService) {
+  type CountryGroupPrices = Map[FulfilmentOptions, Map[ProductOptions, Map[BillingPeriod, List[PriceSummary]]]]
+  type ProductPrices = Map[CountryGroup, CountryGroupPrices]
+  type GroupedPriceList = Map[(FulfilmentOptions, ProductOptions, BillingPeriod), List[PriceSummary]]
+
+  def getPrices[T <: Product](product: T, maybePromoCode: Option[PromoCode]): ProductPrices =
+    product.supportedCountries.map(
+      countryGroup =>
+        countryGroup -> getPricesForCountryGroup(product, countryGroup, maybePromoCode)
+    ).toMap
+
+  def getPricesForCountryGroup[T <: Product](product: T, countryGroup: CountryGroup, maybePromoCode: Option[PromoCode]): CountryGroupPrices = {
```

Comment:
There is some pretty dense code in this class, basically it is taking the flat structures of product rate plans and prices applying any promotions to those prices and then nesting in the following structure:
CountryGroup -> FulfilmentOptions -> ProductOptions -> BillingPeriod